src/Lykke.Common.Validation/NoSpecialCharacters/NoSpecialCharactersValidator.cs

```diff
@@ -1,23 +1,67 @@
-﻿using System.Linq;
+﻿using System;
+using System.Linq;
 
 namespace Lykke.Common.Validation.NoSpecialCharacters
 {
     /// <summary>
     ///     Validates string does not contain special characters:
-    ///     !@#$%^&amp;*()-_=+:;.,"'\/?&lt;&gt;|~[]{}`
     /// </summary>
     public class NoSpecialCharactersValidator
     {
-        private static readonly char[] RestrictedCharacters =
+        /// <summary>
+        ///     Set of default restricted characters.
+        /// </summary>
+        private static readonly char[] DefaultRestrictedCharacters =
         {
             '!', '@', '#', '$', '%', '^', '&', '*', '(', ')', '-', '+', '_', '=', ':', ';', '.', ',', '"', '\'', '\\',
             '/', '|', '?', '<', '>', '~', '[', ']', '{', '}', '`'
         };
 
         /// <summary>
-        ///     Validates string does not contain special characters:
+        ///     Set of characters to be validated for.
+        ///     If input string contains any of these characters, validation result would be
+        ///     negative.
+        ///     By default, equals to <see cref="DefaultRestrictedCharacters" />.
+        ///     Could be modified with additional allowed and restricted characters through constructor with configuration.
+        /// </summary>
+        private readonly char[] _restrictedCharacters;
+
+        /// <summary>
+        ///     Constructs validator with default set of restricted characters.
         ///     !@#$%^&amp;*()-_=+:;.,"'\/?&lt;&gt;|~[]{}`
         /// </summary>
+        public NoSpecialCharactersValidator()
+        {
+            _restrictedCharacters = DefaultRestrictedCharacters;
+        }
+
+        /// <summary>
+        ///     Constructs validator with custom set of allowed and restricted characters.
+        /// </summary>
+        /// <param name="configAction">
+        ///     Delegate to configure validator. Used for setting allowed characters and add additional restricted characters.
+        /// </param>
+        /// <exception cref="ArgumentNullException">Thrown when <paramref name="configAction" /> is null.</exception>
+        /// <exception cref="ArgumentException">Thrown if you set same characters for allowed and restricted.</exception>
+        public NoSpecialCharactersValidator(Action<INoSpecialCharactersConfigurationExpression> configAction) : this()
```

Comment:
Maybe it would be simpler if we will completely override defaults, when configuration is used?

Reply:
The main idea of NoSpecialCharactersValidator is to validate for characters that could be used in scripts.
From my point of view, it makes no sense. I made the configuration to allow users to slightly modify the base set of restricted characters for specific cases. If we allow to completely override default values it would be more difficult for them to add all the symbols again. For this needs, we need a new validator I think, something like WithoutSpecificCharacters.